app/models/equipment_model.rb

```diff
@@ -197,6 +197,15 @@ def num_available(start_date, due_date)
     num_available_from_source(start_date, due_date, relevant_reservations)
   end
 
+  def num_available_excluding(start_date, due_date, exclude_reservation)
+    # for if you just want the number available, 1 query to get
+    # relevant reservations
+    relevant_reservations = Reservation.for_eq_model(self).finalized
+                            .reserved_in_date_range(start_date, due_date)
+                            .all.reject { |r| r.id == exclude_reservation.id }
```

Comment:
Rather than use `reject` you should be able to use `where.not(id: excluded_reservation.id)` so that it's part of the query instead of a separate Array operation. You could also therefore pass in an array of reservations to exclude and call `where.not(id: excluded_reservatons.map(&:id)`.